src/main/java/com/gooddata/GoodDataEndpoint.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2007-2016, GoodData(R) Corporation. All rights reserved.
+ */
+package com.gooddata;
+
+import org.apache.http.HttpHost;
+
+import static com.gooddata.util.Validate.notEmpty;
+
+/**
+ * GoodData Platform endpoint represented by host, port and protocol
+ */
+public class GoodDataEndpoint {
+
+    private static final String PROTOCOL = "https";
```

Comment:
I can either create cyclic dependency between those classes and use constants from GoodData or remove constructors with default values in this class. Or we can just live with little duplication...